game/src/game/player.rs

```diff
@@ -61,6 +132,563 @@ fn setup_player(q: Query<(&Transform, Entity), Added<PlayerBlueprint>>, mut comm
             },
             animation: Default::default(),
         },));
+        println!("player spawned")
+    }
+}
+
+struct PlayerStatePlugin;
+
+impl Plugin for PlayerStatePlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            Update,
+            (
+                // test_trans_int,
+                // transition::<Idle, Grounded>,
+                transition::<Falling, Grounded>,
+                transition::<Grounded, Falling>,
+                // transition::<Falling, Jumping>,
+                transition::<Falling, Running>,
+                transition::<Grounded, Jumping>,
+                transition::<Jumping, Falling>,
+                transition::<Idle, Running>,
+                transition::<Running, Idle>,
```

Comment:
If we are going to have so many systems that typically do nothing on most frames, they better have run conditions. Otherwise significant perf overhead can accumulate from them simply existing (and bevy having to schedule and spawn tasks to run them).

This also applies to other player systems, since you are taking the approach of splitting player movement mechanics into many different systems per-state. They should have RCs to check if the player entity is in the right state. RCs are cheap and running systems (even if they do nothing / early return) has a lot more overhead. RCs are your friend. :)

Reply:
That sounds good, using conditions like `any_with_component` or `any_component_removed`?